storage/store/inmemory/unsynchronized/registers.go

```diff
@@ -0,0 +1,114 @@
+package unsynchronized
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/bits-and-blooms/bloom/v3"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type RegisterEntries map[flow.RegisterID]flow.RegisterValue
+type HeightToRegisterEntries map[uint64]RegisterEntries
+
+type Registers struct {
+	firstHeight  uint64
+	latestHeight uint64
+	lock         sync.RWMutex
+	store        HeightToRegisterEntries
+	bloomFilter  *bloom.BloomFilter
+}
+
+var _ storage.RegisterIndex = (*Registers)(nil)
+
+func NewRegisters(
+	firstHeight uint64,
+	latestHeight uint64,
+	estimatedNumberOfRegisters uint,
+	falsePositiveRate float64,
+) *Registers {
+	return &Registers{
+		firstHeight:  firstHeight,
+		latestHeight: latestHeight,
+		store:        make(HeightToRegisterEntries),
+		bloomFilter:  bloom.NewWithEstimates(estimatedNumberOfRegisters, falsePositiveRate),
+	}
+}
+
+// Get register by the register ID at a given block height.
+//
+// If the register at the given height was not indexed, returns the highest
+// height the register was indexed at.
+// Expected errors:
+// - storage.ErrHeightNotIndexed if the given height was not indexed yet or lower than the first indexed height.
+// - storage.ErrNotFound if the given height is indexed, but the register does not exist.
+func (r *Registers) Get(ID flow.RegisterID, height uint64) (flow.RegisterValue, error) {
```

Comment:
nit: avoid using variable names that start with a capital letter for function args
```suggestion
func (r *Registers) Get(registerID flow.RegisterID, height uint64) (flow.RegisterValue, error) {
```